engine/v2/database/repository/operations.go

```diff
@@ -98,27 +98,38 @@ func mapOperations(operations iter.Seq[*txmodels.NewOperation]) []database.Opera
 	})
 }
 
-func mapTransaction(transaction *txmodels.NewOperation) *database.TrackedTransaction {
+func mapTransaction(operation *txmodels.NewOperation) *database.TrackedTransaction {
 	tx := &database.TrackedTransaction{
-		ID:       transaction.Transaction.ID,
-		TxStatus: string(transaction.Transaction.TxStatus),
+		ID:       operation.Transaction.ID,
+		TxStatus: string(operation.Transaction.TxStatus),
+		BeefHex:  operation.Transaction.BEEFHex(),
+		RawHex:   operation.Transaction.RawHex(),
 	}
 
-	for _, input := range transaction.Transaction.Inputs {
+	if !operation.Transaction.HasTransactionInputSources() {
```

Comment:
do we really need this? 
If the input sources list is empty, the loop won't run, so this check seem to be ambiguous.